Bluejay/Bluejay/Error.swift

```diff
@@ -8,161 +8,24 @@
-            domain: "Bluejay",
-            code: 8,
-            userInfo: [NSLocalizedDescriptionKey: "Cancelled."]
-        )
-    }
-    
-    static func listenTimedOut() -> NSError {
-        return NSError(
-            domain: "Bluejay",
-            code: 9,
-            userInfo: [NSLocalizedDescriptionKey: "Listen timed out."]
-        )
-    }
-    
-    static func readFailed() -> NSError {
-        return NSError(
-            domain: "Bluejay",
-            code: 10,
-            userInfo: [NSLocalizedDescriptionKey: "Read failed."]
-        )
-    }
-    
-    static func writeFailed() -> NSError {
-        return NSError(
-            domain: "Bluejay",
-            code: 11,
-            userInfo: [NSLocalizedDescriptionKey: "Write failed."]
-        )
-    }
-    
-    static func missingData() -> NSError {
-        return NSError(
-            domain: "Bluejay",
-            code: 12,
-            userInfo: [NSLocalizedDescriptionKey: "No data from peripheral."]
-        )
-    }
-    
-    static func dataOutOfBounds(start: Int, length: Int, count: Int) -> NSError {
-        return NSError(
-            domain: "Bluejay",
-            code: 13,
-            userInfo: [NSLocalizedDescriptionKey: "Cannot extract data with a size of \(count) using start: \(start), length: \(length)."]
-        )
-    }
-        
-    static func unexpectedPeripheral(_ peripheral: PeripheralIdentifier) -> NSError {
-        return NSError(
-            domain: "Bluejay",
-            code: 14,
-            userInfo: [NSLocalizedDescriptionKey: "Unexpected peripheral: \(peripheral.uuid)."]
-        )
-    }
-    
-    static func allowDuplicatesInBackground() -> NSError {
-        return NSError(
-            domain: "Bluejay",
-            code: 15,
-            userInfo: [NSLocalizedDescriptionKey: "Scanning with allow duplicates while in the background is not supported."]
-        )
-    }
-    
-    static func missingServiceIdentifiersInBackground() -> NSError {
-        return NSError(
-            domain: "Bluejay",
-            code: 16,
-            userInfo: [NSLocalizedDescriptionKey: "Scanning without specifying any service identifiers while in the background is not supported."]
-        )
-    }
-    
-    static func backgroundTaskRunning() -> NSError {
-        return NSError(
-            domain: "Bluejay",
-            code: 17,
-            userInfo: [NSLocalizedDescriptionKey: "Regular Bluetooth operation is not available when a background task is running. For reading, writing, and listening, please use only the API found in the Synchronized Peripheral provided to you when working inside a background task block."]
-        )
-    }
-    
-    static func multipleBackgroundTask() -> NSError {
-        return NSError(
-            domain: "Bluejay",
-            code: 18,
-            userInfo: [NSLocalizedDescriptionKey: "Multiple background task is not supported."]
-        )
-    }
-    
+public enum BJError {
```

Comment:
I think we need to not call the error class BJError. Doing the Objective-C style namespacing-via-acronyms doesn't really fit Swift style, and  "BJ" is a particularly unfortunate acronym to use. I don't think we could get away with just calling it "Error", since then anyone importing Bluejay would have to disambiguate Swift.Error a lot. Probably the easiest thing to do is just change the name of the class to "BluejayError", but I'd totally be open to other names.

Reply:
lol 😂😂😂, cannot unsee it anymore

Reply:
Done